web/src/engine/websites/MangaDoom.ts

```diff
@@ -0,0 +1,43 @@
+import { Tags } from '../Tags';
+import icon from './MangaDoom.webp';
+import { DecoratableMangaScraper, Manga, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchCSS, FetchRequest } from './../FetchProvider';
+
+function ChapterInfoExtractor(anchor: HTMLAnchorElement) {
+    const id = anchor.pathname + '/all-pages';
+    const title = anchor.querySelector('span.val').textContent.replace(/\s*-/, '').trim();
+    return { id, title };
+}
+
+@Common.MangaCSS(/^https?:\/\/www\.mngdoom\.com/, 'h5.widget-heading')
+@Common.ChaptersSinglePageCSS('div#chapter_list ul.chapter-list li a', ChapterInfoExtractor )
+@Common.PagesSinglePageCSS('div.inner-page img.img-responsive')
+@Common.ImageDirect()
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('mangadoom', `MangaDoom`, 'https://www.mngdoom.com', Tags.Language.English, Tags.Media.Manga, Tags.Media.Manhua, Tags.Media.Manhwa, Tags.Source.Aggregator);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangalist = [];
+        const paths = [''].concat('abcdefghijklmnopqrstuvwxyz'.split(''));
+        for (const path of paths) {
+            mangalist.push(...await this.getMangasFromCategory(path, provider));
```

Comment:
Is it possible to use existing extension functions from the decorators e.g., `Common.FetchMangasSinglePageCSS(this, provider, '/manga-directory/' + path, 'div.content ul.manga-list li a.manga-info-qtip')` instead of `getMangasFromCategory`?